### ISSUE TITLE
bugzilla queries processed as successful even if server returns error (e.g. 504), should cause failure

### ISSUE THREAD

**Issue by Archaeopteryx** [mozilla/libmozdata#189]
The following function caused `504` gateway timeouts on Bugzilla but got processed like it was successful.
```
def my_function(label, start_date, end_date):
    def bug_handler(bug_data):
         # do something

    fields = [
              'id',
              'product',
              'component',
              'severity',
              'history',
             ]

    params = {
        'include_fields': fields,
        'product': ['Core', 'Firefox', 'Toolkit'],
        'f1': 'bug_group',
        'o1': 'notsubstring',
        'v1': 'security',
        'f2': 'keywords',
        'o2': 'nowords',
        'v2': 'crash',
        'f3': 'OP',
        'j3': 'OR',
        'f4': 'OP',
        'j4': 'AND_G',
        'f5': 'bug_severity',
        'o5': 'changedfrom',
        'v5': 'S1',
        'f6': 'bug_severity',
        'o6': 'changedafter',
        'f7': 'bug_severity',
        'o7': 'changedbefore',
        'f9': 'CP',
        'f10': 'OP',
        'j10': 'AND_G',
        'f11': 'bug_severity',
        'o11': 'changedfrom',
        'v11': 'S2',
        'f12': 'bug_severity',
        'o12': 'changedafter',
        'f13': 'bug_severity',
        'o13': 'changedbefore',
        'f15': 'CP',
        'f16': 'CP',
    }

    params['v6'] = start_date
    params['v7'] = end_date
    params['v12'] = start_date
    params['v13'] = end_date

    bugs_data = []

    Bugzilla(params,
             bughandler=bug_handler,
             timeout=960).get_data().wait()
    data = [bug_data['id'] for bug_data in bugs_data]

    return data
```

**Comment by marco-c** [comment]
We had a similar issue, #185, that was fixed in #186.
Are you using the latest version of the library? If so, there might be other places where we have to add `raise_for_status`.

**Comment by Archaeopteryx** [comment]
Still observing this after an update from 0.1.72 to 0.1.73.
